packages/compiler-core/src/transforms/vSlot.ts

```diff
@@ -201,11 +207,13 @@ export function buildSlots(
       hasDynamicSlots = true
     }
 
-    const slotFunction = buildSlotFn(slotProps, slotChildren, slotLoc)
+    vFor = findDir(slotElement, 'for')
+    if (vFor) vForProps = vFor.exp
```

Comment:
If vForProps is set on line 154 and not set here the old one will be used in line 212
Also this is in a loop so all previously set values of vForProp will be reused in subsequent passes

Reply:
uh~ thanks.